kadmin/server.c

```diff
@@ -329,6 +330,12 @@ kadmind_dispatch(void *kadm_handlep, krb5_boolean initial,
 	    && krb5_principal_compare (contextp->context, contextp->caller,
 				       princ))
 	{
+	    ret = 0;
```

Comment:
Wouldn't it be better to just reverse the sense of the if condition and then drop this `ret = 0; } else {` bit?